docs/how-to/custom-framing.md

```diff
@@ -29,11 +32,26 @@ For more information on working with subtopologies, see the [Subtopologies Guide
 
 To implement a custom framing protocol in F´, will need to implement the following:
 - **Framer**: A component that wraps payload data into frames for transmission.
-- **Deframer**: A component that unpacks received frames, extracts the payload data, and validate the frame.
-- **FrameDetector** (optional): A helper class that detects the start and end of frames in a stream of data, if your transport is stream-based (e.g., TCP, UART).
+- **Deframer**: A component that unpacks received frames, extracts the payload data, and validates the frame.
+- **FrameDetector** Helper Class (optional): A C++ helper class (not an FPP component) that detects the start and end of frames in a stream of data, if your transport is stream-based (e.g., TCP, UART).
+
+> [!TIP]
+> When creating framer/deframer components using `fprime-util new --component`, the recommended settings are to use passive components with events enabled.
 
 The following examples will walk through the implementation of a custom framer and deframer for a hypothetical **MyCustomProtocol** protocol. Implementation details are left to the reader, but examples of such implementations can be found in the [fprime-examples repository](https://github.com/nasa/fprime-examples/tree/devel/FlightExamples/CustomFraming), or within the F´ codebase itself ([Svc.FprimeFramer](../../Svc/FprimeFramer/docs/sdd.md) and [Svc.FprimeDeframer](../../Svc/FprimeDeframer/docs/sdd.md)).
 
+### Pitfalls and Best Practices
+
+Before implementing, consider these best practices:
+
+1. **Centralize Protocol Constants**: Define protocol constants (e.g., start word, header structure, field sizes) in a dedicated `.fpp` types file (e.g., `Types/Types.fpp`). This allows constants like start words, field types, or spacecraft IDs to be reused consistently across your framer, deframer, and frame detector.
+
+2. **Endianness Convention**: F´ serializes all integer types in big-endian (network byte order) as defined in `Fw::Serializable`. If your custom protocol requires little-endian fields, you must explicitly swap bytes when building or parsing frames. This is critical for interoperability with ground systems or other flight software.
```

Comment:
You don't have to swap bytes anymore....but you do have to supply the endinanness flag to the serialize/deserialize calls.